### ISSUE TITLE
bug: React IonModal component doesn't support documented events

### ISSUE THREAD

**Issue by aub** [ionic-team/ionic#19923]
# Bug Report

**Ionic version:**

[x] **4.11.5

**Current behavior:**
The [docs for IonModal](https://ionicframework.com/docs/api/modal) list four events it supports. Of these, the only one that's actually available based on the types is `onDidDismiss`.

**Expected behavior:**
The types, and the implementation support all four events.

**Steps to reproduce:**
Create an app that tries to use the `onDidPresent` event as a prop on an `IonModal`, it will fail when TS does its compile step.

**Related code:**
The props for `IonModal` come from `createOverlayComponent`, which uses `ReactOverlayProps` to define some things. The type for `ReactOverlayProps` supports `onDidDismiss` but none of the other events.

Relevant code [here](https://github.com/ionic-team/ionic/blob/033abe994be17d927d38df2c931010d2b10b9250/packages/react/src/components/createOverlayComponent.tsx#L12) and [here](https://github.com/ionic-team/ionic/blob/930b271a4abf680b08e6755644cece4b95053f15/packages/react/src/components/IonModal.tsx).

**Other information:**

**Ionic info:**

```
Ionic:

   Ionic CLI       : 5.4.4 (/usr/local/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 4.11.5

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 1.3.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v12.12.0 (/usr/local/Cellar/node/12.12.0/bin/node)
   npm    : 6.11.3
   OS     : macOS Catalina
```

**Comment by alekseykarpenko** [comment]
Same issue here, seems to be *critical* when trying to update `IonSlides` inside `IonModal` after closing and opening it again.

**Comment by alekseykarpenko** [comment]
@elylucas any plans or thoughts on this?

**Comment by elylucas** [comment]
@alekseykarpenko @aub 

I'm taking a look. at this today

**Comment by elylucas** [comment]
fixed in 4.11.9
